test/replication-luatest/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+set(TEST_SUITE_NAME "replication-luatest")
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target
+# <replication-luatest-deps> as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS "${TEST_SUITE_NAME};regression;"
+)
+
+list(APPEND TEST_ENV
+  "LUA_PATH=${LUATEST_LUA_PATH}"
+  "VARDIR=${VARDIR}"
+)
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *_test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  add_test(NAME ${test_title}
+           COMMAND ${TARANTOOL_BIN}
+                   # Disable stdout buffering,
+                   # see https://github.com/tarantool/test-run/issues/119.
+                   -e "io.stdout:setvbuf('no')"
+                   ${LUATEST_COMMAND_WITH_FLAGS}
+                   ${test_title}
+           WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+  )
+  set_tests_properties(${test_title} PROPERTIES
+    ENVIRONMENT "${TEST_ENV}"
+    LABELS "${TEST_SUITE_NAME};regression;"
+    DEPENDS ${TEST_SUITE_NAME}-deps
+    FAIL_REGULAR_EXPRESSION "${CTEST_FAIL_REGEXP}"
+  )
+endforeach()
+
+# See test/replication-luatest/suite.ini.
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set_tests_properties(
+    test/replication-luatest/gh_5295_split_brain_test.lua
+    test/replication-luatest/gh_6033_box_promote_demote_test.lua
```

Comment:
For some reason this test fails even in debug build:
```
1355: not ok 6  gh-6033-box-promote-demote.test_simultaneous
1355: #   Instance with replica id 0 was promoted first
1355: #   stack traceback:
1355: #         .../replication-luatest/gh_6033_box_promote_demote_test.lua:113: in function <.../replication-luatest/gh_6033_box_promote_demote_test.lua:112>
1355: #         .../replication-luatest/gh_6033_box_promote_demote_test.lua:112: in function <.../replication-luatest/gh_6033_box_promote_demote_test.lua:108>
1355: #   artifacts:
1355: #         server_2 -> /tmp/t/artifacts/rs-hJSMTrgtyCQu/server_2-AABJ6-TXtS3b
1355: #         server_1 -> /tmp/t/artifacts/rs-hJSMTrgtyCQu/server_1-9ebIlRvp8tIv

```
Run it via the command `ctest --verbose -R  gh_6033_box_promote_demote_test`.

Reply:
Everything is fine on my computer:

<details>
  <summary>log</summary>

```
UpdateCTestConfiguration  from :/home/sergeyb/sources/MRG/tarantool/build/DartConfiguration.tcl
Parse Config file:/home/sergeyb/sources/MRG/tarantool/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/sergeyb/sources/MRG/tarantool/build/DartConfiguration.tcl
Parse Config file:/home/sergeyb/sources/MRG/tarantool/build/DartConfiguration.tcl
Test project /home/sergeyb/sources/MRG/tarantool/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1624
    Start 1624: test/replication-luatest/gh_5295_split_brain_test.lua

1624: Test command: /home/sergeyb/sources/MRG/tarantool/build/src/tarantool "-e" "io.stdout:setvbuf('no')" "/home/sergeyb/sources/MRG/tarantool/test-run/lib/luatest/bin/luatest" "--no-clean" "--verbose" "--no-capture" "--output" "tap" "test/replication-luatest/gh_5295_split_brain_test.lua"
1624: Environment variables: 
1624:  LUA_PATH=/home/sergeyb/sources/MRG/tarantool/?.lua;/home/sergeyb/sources/MRG/tarantool/?/init.lua;/home/sergeyb/sources/MRG/tarantool/test-run/lib/checks/?.lua;/home/sergeyb/sources/MRG/tarantool/test-run/lib/luatest/?/init.lua;/home/sergeyb/sources/MRG/tarantool/test-run/lib/luatest/?.lua;;
1624:  VARDIR=/tmp/t
1624: Test timeout computed to be: 1500
1624: Tarantool version is 3.3.0-entrypoint-208-g18948060f4
1624: TAP version 13
1624: 1..12
1624: # Started on Tue Nov 19 16:44:07 2024
1624: # Starting group: gh-5295
1624: ok     1	gh-5295.test_async_old_term
1624: ok     2	gh-5295.test_bad_confirm_old_term
1624: ok     3	gh-5295.test_good_confirm_old_term
1624: ok     4	gh-5295.test_bad_rollback_old_term
1624: ok     5	gh-5295.test_good_rollback_old_term
1624: ok     6	gh-5295.test_demote_same_term
1624: ok     7	gh-5295.test_promote_same_term
1624: ok     8	gh-5295.test_promote_new_term_small_lsn
1624: ok     9	gh-5295.test_promote_new_term_big_lsn
1624: ok     10	gh-5295.test_promote_new_term_conflicting_queue
1624: # Starting group: test-confirm-very-old-term
1624: ok     11	test-confirm-very-old-term.test_confirm
1624: ok     12	test-confirm-very-old-term.test_promote_split_brain_without_data
1624: # Ran 12 tests in 10.338 seconds, 12 succeeded, 0 failed
1/1 Test #1624: test/replication-luatest/gh_5295_split_brain_test.lua ...   Passed   10.51 sec

The following tests passed:
	test/replication-luatest/gh_5295_split_brain_test.lua

100% tests passed, 0 tests failed out of 1

Label Time Summary:
regression             =  10.51 sec*proc (1 test)
replication-luatest    =  10.51 sec*proc (1 test)

Total Test time (real) =  10.56 sec
```

</details>

`-DCMAKE_BUILD_TYPE=Debug` was used for CMake configuration.

Tarantool 3.3.0-entrypoint-208-g18948060f4
Target: Linux-x86_64-Debug
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=TRUE
Compiler: Clang-20.0.0
C_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/home/sergeyb/sources/MRG/tarantool=. -std=c11 -Wall -Wextra -Wno-gnu-alignof-expression -Wno-cast-function-type -Werror -g -ggdb -O0 
CXX_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/home/sergeyb/sources/MRG/tarantool=. -std=c++11 -Wall -Wextra -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type -Werror -g -ggdb -O0

Reply:
Never mind, it is just a [flaky test](https://mon.tarantool.io/grafana/d/vQ35ub74k/test-statistics?orgId=1&var-bucket=multivac-tests&var-bucket_table=multivac-table&var-job_name=All&var-test_name=replication-luatest%2Fgh_6033_box_promote_demote%3E&var-branch=All&var-os_version=All&var-architecture=All&var-gc64=All&var-compiler_version=All&var-libc_version=All&var-debug=All&var-configuration=All&var-test_type=All&var-test_subtype=All&var-test_attempt=All&var-period=1w&var-tag=_start). Rerun works fine for me (unlucky coincidence that this test failed twice in a raw on the local run).
Also, you may try to reproduce it with `--repeat-until-fail 100`.